### ISSUE TITLE
apply-setters function does not apply substitutions

### ISSUE THREAD

**Issue by johnbelamaric** [GoogleContainerTools/kpt#1931]
### Expected behavior
I would expect the apply-setters function to work in a manner identical to `kpt cfg set`.

### Actual behavior
Substitutions are ignored, which makes them useless if you want to use the package in an automation pipeline with apply-setters.

### Information

Kpt Version: 0.39.2

### Steps to reproduce the behavior
Run apply-setters against a package with substitutions defined. Only the setters are applied.

**Comment by mikebz** [comment]
@phanimarupaka can you please take a look at this one using kpt `next` branch?

**Comment by mikebz** [comment]
@johnbelamaric just to be clear you have tried using this with kpt `next` build or you tried the new function with old kpt?

https://catalog.kpt.dev/apply-setters/v0.1/

Here is the example:
```
apiVersion: v1
kind: Deployment
metadata:
  name: nginx-deployment # kpt-set: ${image}-deployment
spec:
  replicas: 1 # kpt-set: ${replicas}
```
where there is a substitution.  Is this the thing that you tried and it's not working for you?

**Comment by phanimarupaka** [comment]
> ### Expected behavior
> I would expect the apply-setters function to work in a manner identical to `kpt cfg set`.
> 
Yes. It does.
> ### Actual behavior
> Substitutions are ignored, which makes them useless if you want to use the package in an automation pipeline with apply-setters.
> 
We are getting rid of the concept of substitutions and the pattern will just be inline comment. So the current version of setters and substitutions are not compatible with apply-setters function. However, we will be releasing a new image `gcr.io/kpt-fn/fix`  (WIP) which will fix the comment pattern in your existing kpt package to the new pattern so that apply-setters function can work just fine. 

In short, you have to fix your package for apply-setters to work.
> ### Information
> Kpt Version: 0.39.2
> 
> ### Steps to reproduce the behavior
> Run apply-setters against a package with substitutions defined. Only the setters are applied.

**Comment by johnbelamaric** [comment]
Ah, I see. So the newer version of `kpt` does things differently, but supports essentially the same functionality?

**Comment by phanimarupaka** [comment]
Yes it does, in a more cleaner fashion.

**Comment by johnbelamaric** [comment]
Ok, awesome. Is there a released version of `kpt` that has the new behavior? Or do I need to build it?

**Comment by mikebz** [comment]
@johnbelamaric we are going to open up the new build for people to give us feedback in a few weeks.  Right now we know we are missing some P1 docs and P1 features.  If you have deadlines that press you to get a build early and start migrating content - let us know. If not we will welcome your feedback in M3 milestone - https://github.com/GoogleContainerTools/kpt/milestones

**Comment by mikebz** [comment]
For right now I am going to close this, as it's not a bug and we have the scenario supported in the `next` branch.
